manifests/config.pp

```diff
@@ -1,58 +1,68 @@
+# Class: check_mk::config
+#
+# Configures check_mk.
+# 
+# @summary Configures check_mk. Should not be called directly.
+# 
 
 class check_mk::config (
-  $site,
-  $host_groups = undef,
+  String $site,
+  Optional[Hash]  $host_groups     = undef,
+  Optional[Array] $all_hosts_static = undef,
```

Comment:
How about `Array $all_hosts_static = []`?  You could then simplify the template (get rid of the `<% if static_hosts.is_a? Array -%>`).  I'm also fine with changing this is a separate PR.

Reply:
Lets fix the templates in a later pull request. This is not the only template that might be simplified (or even removed).